web/app/sankey/page.tsx

```diff
@@ -1,54 +1,146 @@
+    }
+    return false;
+  });
+  const [showStatus, setShowStatus] = useState<boolean>(() => {
+    if (typeof window !== 'undefined') {
+      return localStorage.getItem('sankeyShowStatus') === 'true';
+    }
+    return false;
+  });
+  
+  // Global state
+  const { filterByType, paused, setPaused } = useGlobalDataStream();
+  const { setMenuContent } = useGlobalMenu();
   
   // Get only Stratum V1 data for this visualization
   const stratumV1Data = useMemo(() => {
     return filterByType(StreamDataType.STRATUM_V1);
   }, [filterByType]);
+  
+  // Calculate dynamic height based on pool count using square root scaling
+  const dynamicHeight = useMemo(() => {
+    const calculateHeight = (poolCount: number): number => {
+      const baseHeight = 200;
+      const maxHeight = 1400;
+      const multiplier = 190; // Calculated to reach ~1400px at 40 pools with baseHeight = 200
+      const calculatedHeight = baseHeight + (Math.sqrt(poolCount) * multiplier);
+      return Math.min(calculatedHeight, maxHeight);
+    };
+    
+    // Count unique pools from the data
+    const poolCount = new Set(
+      stratumV1Data
+        .filter(item => item.type === StreamDataType.STRATUM_V1)
+        .map(item => item.data.pool_name)
+    ).size;
+    return calculateHeight(Math.max(poolCount, 1)); // Ensure minimum of 1 pool for calculation
+  }, [stratumV1Data]);
+  
+  // Wrap onDataRendered in useCallback to prevent unnecessary re-renders
+  const handleDataRendered = useCallback((nodes: number, links: number) => {
+    setNodeLinkCounts({ nodes, links });
+  }, []);
+  
+  // Persist showStatus and showRawEvents to localStorage
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('sankeyShowStatus', String(showStatus));
+      localStorage.setItem('sankeyShowRawEvents', String(showRawEvents));
+    }
+  }, [showStatus, showRawEvents]);
 
+  
+  // Set the menu content when the component mounts
+  useEffect(() => {
+    setMenuContent(
+      <SankeyMenu
+        paused={paused}
+        setPaused={setPaused}
+        showSettings={showSettings}
+        setShowSettings={setShowSettings}
+        showLabels={showLabels}
+        setShowLabels={setShowLabels}
+        showStatus={showStatus}
+        setShowStatus={setShowStatus}
+        showRawEvents={showRawEvents}
+        setShowRawEvents={setShowRawEvents}
+      />
+    );
+    return () => setMenuContent(null);
+  }, [paused, showSettings, showLabels, showStatus, showRawEvents, setMenuContent, setPaused]);
+  
   return (
     <main className="flex min-h-screen flex-col items-center p-8">
       <div className="w-full max-w-7xl">
-        <h1 className="text-2xl font-bold mb-6">Stratum V1 Data Stream</h1>
-        
-        {/* Sankey Menu for controls */}
-        <div className="mb-4">
-          <SankeyMenu />
+        <h1 className="text-2xl font-bold mb-6">Mining Data Flow Visualization</h1>
```

Comment:
I think we should change the title to something related to merkle branches. Maybe something like:

```suggestion
        <h1 className="text-2xl font-bold mb-6">Sankey Merkle Branches</h1>
```
This is a nit and we can leave it as-is if you want, but thinking it should at least mention merkle branches.